src/main/java/org/opentripplanner/routing/api/request/request/RouteViaRequest.java

```diff
@@ -1,16 +1,49 @@
 package org.opentripplanner.routing.api.request.request;
 
+import java.util.ArrayList;
 import java.util.List;
-
-//  User request: from/to time/location; preferences slack/cost/reluctance
+import org.opentripplanner.routing.api.request.RouteRequest;
+import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
 
 // TODO VIA: Javadoc
 public class RouteViaRequest {
 
-  List<ViaLocation> viaPoints;
+  private final RoutingPreferences preferences;
+
+  private final RouteRequest routeRequest;
+
+  private final List<ViaLeg> viaLegs = new ArrayList<>();
+
+  public RouteViaRequest(
+    RoutingPreferences preferences,
+    List<ViaLocation> viaLocations,
+    List<JourneyRequest> viaJourneys,
+    RouteRequest routeRequest
```

Comment:
We should not have a `RouteRequest` as input parameter, but instead we should add fields which we can use to create one at runtime